### DISCUSSION TITLE
How to download micropython 1.21.0 ?

### DISCUSSION THREAD

**Discussion by patfrench** [micropython/micropython#13771] [General]
Hello,

I'd like to build a version of micropython 1.21.0
I've downloaded the zip file from sourceforge but I'm having problems building it:

or maybe there is another solution ?

make -C mpy-cross
cd port/rp2
make submodules : error => parent directory is not a git repository

Thanks in advance

**Comment by patfrench** [comment]
I succeeded:
git init
git add .
and then ok

On the other hand, now that I'm trying to migrate from evolutions 1.19 to 1.21 I always get the same error message:

```
CMake error at CmakeLists.txt:83 (add_executable):
cannot find source file:
/src/tusb.c
```

I admit I'm not a CMAKE expert 

thanks in advance

**Reply by robert-hh** [reply]
The most reliable way to build a firmware is starting with cloning the whole MicroPython github repository. Then you can run `make submodules`. After that, you have the full set of files. And you have access to every version of MicroPyhton.

**Reply by patfrench** [reply]
Hi !
I will try
Thank you robert-hh for your support
